core-processor/src/ext.rs

```diff
@@ -883,42 +883,74 @@ impl Ext {
         pages: WasmPageNumber,
         mem: &mut impl Memory,
     ) -> Result<WasmPageNumber, ProcessorError> {
+        struct AllocGasCharger<'a> {
+            charged: u64,
+            ext: &'a mut Ext,
+        }
+
+        impl<'a> AllocGasCharger<'a> {
+            fn new(ext: &'a mut Ext) -> Self {
+                Self { charged: 0, ext }
+            }
+
+            fn calculate_gas(&self, alloc: u32, mem_grow: u32) -> u64 {
+                let alloc = alloc as u64;
+                let mem_grow = mem_grow as u64;
+                0_u64
+                    .saturating_add(alloc.saturating_mul(self.ext.context.config.alloc_cost))
+                    .saturating_add(mem_grow.saturating_mul(self.ext.context.config.mem_grow_cost))
```

Comment:
```suggestion
                    alloc
                    .saturating_mul(self.ext.context.config.alloc_cost)
                    .saturating_add(mem_grow.saturating_mul(self.ext.context.config.mem_grow_cost))
```
is something like that possible/better?

Reply:
It's possible but I doubt it better as it has lower readability